include/openthread/srp_client.h

```diff
@@ -245,6 +260,60 @@ const otSockAddr *otSrpClientGetServerAddress(otInstance *aInstance);
  */
 void otSrpClientSetCallback(otInstance *aInstance, otSrpClientCallback aCallback, void *aContext);
 
+/**
+ * This function enables the auto-start mode.
+ *
+ * This is only available when auto-start feature `OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE` is enabled.
+ *
+ * Config option `OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_MODE` specifies the default auto-start mode (whether
+ * it is enabled or disabled at the start of OT stack).
+ *
+ * When auto-start is enabled, the SRP client will monitor the Thread Network Data for SRP Server Service entries
+ * and automatically start and stop the client when an SRP server is detected.
+ *
+ * If multiple SRP servers are found, a random one will be selected. If the selected SRP server is no longer
+ * detected (not longer present in the Thread Network Data), the SRP client will be stopped and then it may switch
+ * to another SRP server (if available).
```

Comment:
One thing we need to consider is that the DNS-SD client (I believe we want to add `auto-server-discovery` to the DNS-SD client as well) and SRP client should choose the same SRP server when multiple servers are present at the same time. Otherwise, the DNS-SD client may not be able to discover the services registered by the SRP client.

I think we need a srp-server-selecting module/method that can be shared by at least the SRP client and DNS-SD client.

Reply:
In my view, SRP and DND-SD while related are two different protocols and functions. SRP client is used by device to publish its services. DSN-SD client is used to discover services within the home/network. It can be true that we may end up running SRP server and DNS server on same device (on a BR), but we should not assume or rely that they are always on same device (specially from client perspectives). These are two different network entities. 

I think it is the job of a DNS name server to be able to answer questions related to all domains/services it is managing (how it knows and collects the service info, e.g. through interacting with SRP servers or mDNS or ... should not matter to clients that are using it).

That said, we already have API to get the selected server currently being used by SRP client (when auto-start is being used).  So if a use-case wants to assume SRP server is also DNS name server it can be done by next layer code but I don't think it's good idea to add  code/logic within OT itself that assumes this.

Reply:
Agree with both of you. I think the details are yet to be defined by Thread BR Spec.